build.bat

```diff
@@ -507,8 +506,9 @@ REM BUILD function: build various targets
 		rem call:BUILDTYPE
 
 		set CMAKE_ARGS=-DDRIVER_BASE_NAME=%DRIVER_BASE_NAME%
-		REM no explicit x86 generator and is the default (MSVC2017 only?).
-		set CMAKE_ARGS=!CMAKE_ARGS! -DCMAKE_GENERATOR_PLATFORM=%TARCH:x86=%
+		REM CMAKE_GENERATOR_TOOLSET (or cmake -T v142) to specify a toolset
```

Comment:
I don't think this is needed with the `-G "Visual Studio 16 2019"` below, as `v142` of the toolset is the default with Visual Studio 2019.

Reply:
Indeed, using `-G` is just clearer.
I've also changed the script to select 2019 or 2017 if both are installed and in that order.
Hopefully it should now work fine with heterogeneous CI workers.